core/src/main/java/org/elasticsearch/index/query/QueryStringQueryParser.java

```diff
@@ -214,16 +214,16 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (qpSettings.queryString() == null) {
             throw new QueryParsingException(parseContext, "query_string must be provided with a [query]");
         }
-        qpSettings.defaultAnalyzer(parseContext.mapperService().searchAnalyzer());
-        qpSettings.defaultQuoteAnalyzer(parseContext.mapperService().searchQuoteAnalyzer());
+        qpSettings.defaultAnalyzer(parseContext.context().mapperService().searchAnalyzer());
+        qpSettings.defaultQuoteAnalyzer(parseContext.context().mapperService().searchQuoteAnalyzer());
```

Comment:
lots of things here will need to be moved to the data node too once we refactor this query I think